tests/unit/Core/Content/Media/Thumbnail/ThumbnailSizeCalculatorTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Content\Media\Thumbnail;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Media\Aggregate\MediaThumbnailSize\MediaThumbnailSizeEntity;
+use Shopware\Core\Content\Media\Thumbnail\ThumbnailSizeCalculator;
+
+#[CoversClass(ThumbnailSizeCalculator::class)]
+class ThumbnailSizeCalculatorTest extends TestCase
+{
+    /**
+     * @param array<string, int> $imageSize
+     * @param array<string, int> $preferredThumbnailSize
+     * @param array<string, int> $expectedSize
+     */
+    #[DataProvider('thumbnailSizeProvider')]
+    public function testCalculateSize(array $imageSize, array $preferredThumbnailSize, array $expectedSize): void
+    {
+        $thumbnailSizeEntity = new MediaThumbnailSizeEntity();
+        $thumbnailSizeEntity->setWidth($preferredThumbnailSize['width']);
+        $thumbnailSizeEntity->setHeight($preferredThumbnailSize['height']);
+
+        $thumbnailSizeCalculator = new ThumbnailSizeCalculator();
+        $calculatedSize = $thumbnailSizeCalculator->calculateSize($imageSize, $thumbnailSizeEntity);
+
+        static::assertEquals($expectedSize, $calculatedSize);
+    }
+
+    /**
+     * @return array
```

Comment:
```suggestion
     * @return list<array{0: ImageSize, 1: ImageSize, 2: ImageSize}>
```

We have a phpstan rule that all array types need to be typed ;) of course you then have to import the type here as well

Reply:
Done